src/libraries/Microsoft.PowerFx.Json/Public/Types/UserProvidedTypeDefinitions/FormulaTypeSchema.cs

```diff
@@ -56,15 +62,16 @@ public FormulaType ToFormulaType(DefinedTypeSymbolTable definedTypeSymbols, Form
             }
 
             var logicalNameToRecordType = settings.LogicalNameToRecordType;
-            if (typeName == SchemaTypeName.ExpandableTableTypeName.Name && logicalNameToRecordType != null)
-            {
-                return logicalNameToRecordType.Invoke(this.Description).ToTable();
-            }
-            else if (typeName == SchemaTypeName.ExpandableRecordTypeName.Name && logicalNameToRecordType != null)
+            if ((typeName == SchemaTypeName.CustomTableTypeName.Name || typeName == SchemaTypeName.CustomRecordTypeName.Name) && logicalNameToRecordType != null)
```

Comment:
>typeName == SchemaTypeName.CustomTableTypeName.Name || typeName == SchemaTypeName.CustomRecordTypeName.Name [](http://example.com/codeflow?start=17&length=107)

nit: since these type names are the same, can we just do one check?